src/board.py

```diff
@@ -0,0 +1,68 @@
+import networkx as nx
```

Comment:
Interesting didn't know about this dep :+1:

Reply:
It was an arbitrary choice, but it seems to be a good one. Quite easy to use, relatively well documented.